gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -309,9 +322,22 @@ public void start() {
 
     // Start metric reporting
     if (this.containerMetrics.isPresent()) {
```

Comment:
Can this be packaged into a method like `initMetricReporter` ? And `isMetricReportingFailureFatal `, `isEventReportingFailureFatal ` are all local variables that only relevant to this part. We just need to pass `clusterConfig` to init method.

Reply:
Thanks! Addressed.